File: packages/http-client-csharp/generator/Microsoft.TypeSpec.Generator.Input/src/InputTypes/OperationPaging.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System.Collections.Generic;

namespace Microsoft.TypeSpec.Generator.Input
{
    public sealed class OperationPaging
    {
        public IReadOnlyList<string> ItemPropertySegments { get; internal set; }
        public NextLink? NextLink { get; internal set; }
        public ContinuationToken? ContinuationToken { get; internal set; }

        public OperationPaging(IReadOnlyList<string> itemPropertySegments, NextLink? nextLink, ContinuationToken? continuationToken)
        {
            ItemPropertySegments = itemPropertySegments;
            NextLink = nextLink;
            ContinuationToken = continuationToken;
        }
    }
}

File: packages/http-client-csharp/emitter/src/type/operation-paging.ts
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

import { InputOperation } from "./input-operation.js";
import { InputParameter } from "./input-parameter.js";
import { ResponseLocation } from "./response-location.js";

export interface OperationPaging {
  NextLink?: NextLink;
  ContinuationToken?: ContinuationToken;
  ItemPropertySegments: string[];
}

export interface NextLink {
  Operation?: InputOperation;
  ResponseSegments: string[];
  ResponseLocation: ResponseLocation;
}

export interface ContinuationToken {
  Parameter: InputParameter;
  ResponseSegments: string[];
  ResponseLocation: ResponseLocation;
}
